bbq/vm/context.go

```diff
@@ -264,6 +281,43 @@ func (c *Context) GetResourceDestructionContextForLocation(location common.Locat
 	return c
 }
 
+func AttachmentBaseAndSelfValues(
+	c *Context,
+	v *interpreter.CompositeValue,
+	method FunctionValue,
+) (base *interpreter.EphemeralReferenceValue, self *interpreter.EphemeralReferenceValue) {
+	// CompositeValue.GetMethod, as in the interpreter we need an authorized reference to self
+	var unqualifiedName string
+	switch functionValue := method.(type) {
+	case CompiledFunctionValue:
+		parts := strings.Split(functionValue.Function.Name, ".")
+		unqualifiedName = parts[len(parts)-1]
```

Comment:
I think we can update the compiler to only store the un-qualified name in `CompiledFunctionValue.Function.Name` field, so we don't have to do a string splits at runtime.

So far the `bbq.Function.Name` only mattered/used for built-in/native functions, for function look-up in the linker. Unifying it for all types of functions (i.e: having the un-qualified-name always) makes sense, and should be safe.